### ISSUE TITLE
Relative paths for local images

### ISSUE THREAD

**Issue by jelle82** [BookStackApp/BookStack#1137]
Make the paths to locally stored images relative instead of absolute. For example `<img src="/uploads/images/gallery/cat.png" />` instead of `<img src="http://bookstack.example.com/uploads/images/gallery/cat.png" />`,

It increases portability. f you want to change the domain name for the Bookstack App there is no need to change all the image urls afterwards.

**Comment by Sibbo07** [comment]
That would be very helpfull!

**Comment by ssddanbrown** [comment]
This would be nice but will make the logic more complex. 
Was a while ago but I think i decided to use full paths since we'd have to handle full paths anyway for external storage systems (S3). Easier to standardise on one format.

**Comment by ssddanbrown** [comment]
@andreicocari Yeah, Although we essentially already have this via `APP_URL` in the `.env` file.

**Comment by recklessnl** [comment]
+1 for this!

**Comment by comnam90** [comment]
+1 for this, it's needed for environments where the internal and external URLs are different.

**Comment by GarethFox** [comment]
+1 too ^^
I face the same issue using a temporary url to create content and modifying the url when staging to production environment with final url, all the images are not displayed any more, they are referring to the old url in use when uploaded.

**Comment by Technik-J** [comment]
+1
Have the same issue after migrating to another domain.

**Comment by Stefansen** [comment]
any update on this issues?

**Comment by reingit** [comment]
I fixed it like this for now:

1) Backup db: mysqldump bookstack > db.backup.sql
2) Comment out return 1 for url check in app/Console/Commands/UpdateUrl.php
3) php artisan bookstack:update-url https://bookstack.non.tor https://bookstack.onion
4) php artisan bookstack:update-url https://bookstack.onion ""

Havent really tested it much, but everything I needed relative is relative now. Might crontab this in the future.

**Comment by ssddanbrown** [comment]
@GarethFox Just be aware that this can mess with some functionality (like references, ZIP export handling) since the base URL is used as an anchor point for parsing/discovery, and it'd make it tricky to change URLs (especially sub-paths) if needed in the future.

**Comment by ssddanbrown** [comment]
BookStack is now [managed on Codeberg](https://codeberg.org/bookstack/bookstack).

As part of this change, we're closing all existing GitHub issues and pull requests.
You'll instead find this issue on Codeberg here: 
https://codeberg.org/bookstack/bookstack/issues/1137
  
Any further discussion for this will now be done on Codeberg.
Therefore this GitHub thread will be locked.
